das/avail/utils.go

```diff
@@ -0,0 +1,126 @@
+package avail
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	gsrpc "github.com/centrifuge/go-substrate-rpc-client/v4"
+	gsrpc_types "github.com/centrifuge/go-substrate-rpc-client/v4/types"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/vedhavyas/go-subkey"
+)
+
+var localNonce uint32 = 0
+
+func GetAccountNonce(accountNonce uint32) uint32 {
+	if accountNonce > localNonce {
+		localNonce = accountNonce
+		return accountNonce
+	}
+	localNonce++
+	return localNonce
+}
+
+func GetExtrinsicIndex(api *gsrpc.SubstrateAPI, blockHash gsrpc_types.Hash, address string, nonce gsrpc_types.UCompact) (int, error) {
+	// Fetching block based on block hash
+	avail_blk, err := api.RPC.Chain.GetBlock(blockHash)
+	if err != nil {
+		return -1, fmt.Errorf("❌ cannot get block for hash:%v and getting error:%w", blockHash.Hex(), err)
+	}
+
+	// Extracting the required extrinsic according to the reference
+	for i, ext := range avail_blk.Block.Extrinsics {
+		// Extracting sender address for extrinsic
+		ext_Addr, err := subkey.SS58Address(ext.Signature.Signer.AsID.ToBytes(), 42)
+		if err != nil {
+			log.Error("❌ unable to get sender address from extrinsic", "err", err)
+		}
+
+		if ext_Addr == address && ext.Signature.Nonce.Int64() == nonce.Int64() {
+			return i, nil
+		}
+	}
+	return -1, fmt.Errorf("❌ unable to find any extrinsic in block %v, from address %v with nonce %v", blockHash, address, nonce)
+}
+
+type BridgeApiResponse struct {
+	BlobRoot           gsrpc_types.Hash   `json:"blobRoot"`
+	BlockHash          gsrpc_types.Hash   `json:"blockHash"`
+	BridgeRoot         gsrpc_types.Hash   `json:"bridgeRoot"`
+	DataRoot           gsrpc_types.Hash   `json:"dataRoot"`
+	DataRootCommitment gsrpc_types.Hash   `json:"dataRootCommitment"`
+	DataRootIndex      uint64             `json:"dataRootIndex"`
+	DataRootProof      []gsrpc_types.Hash `json:"dataRootProof"`
+	Leaf               gsrpc_types.Hash   `json:"leaf"`
+	LeafIndex          uint64             `json:"leafIndex"`
+	LeafProof          []gsrpc_types.Hash `json:"leafProof"`
+	RangeHash          gsrpc_types.Hash   `json:"rangeHash"`
+}
+
+func QueryMerkleProofInput(blockHash string, extrinsicIndex int, t int64) (MerkleProofInput, error) {
+	// Quering for merkle proof from Bridge Api
```

Comment:
please clean up

Reply:
Done in 154e91e6633633c9ab4fa3227dcfb293e6c8e37f